appserver/payara-appserver-modules/environment-warning/src/main/java/fish/payara/appserver/environment/warning/config/admin/SetEnvironmentWarningConfigurationCommand.java

```diff
@@ -0,0 +1,140 @@
+import fish.payara.appserver.environment.warning.EnvironmentWarningService;
+import fish.payara.appserver.environment.warning.config.EnvironmentWarningConfiguration;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyVetoException;
+import java.util.logging.Logger;
+import javax.inject.Inject;
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.CommandLock;
+import org.glassfish.api.admin.ExecuteOn;
+import org.glassfish.api.admin.RestEndpoint;
+import org.glassfish.api.admin.RestEndpoints;
+import org.glassfish.api.admin.RuntimeType;
+import org.glassfish.config.support.CommandTarget;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.internal.api.Target;
+import org.jvnet.hk2.annotations.Service;
+import org.jvnet.hk2.config.ConfigSupport;
+import org.jvnet.hk2.config.SingleConfigCode;
+import org.jvnet.hk2.config.TransactionFailure;
+
+/**
+ *
+ * @author Matt Gill
+ */
+@Service(name = "set-environment-warning-configuration")
+@PerLookup
+@CommandLock(CommandLock.LockType.NONE)
+@ExecuteOn(RuntimeType.DAS)
+@TargetType(value = {CommandTarget.DAS})
+@RestEndpoints({
+    @RestEndpoint(configBean = Domain.class,
+            opType = RestEndpoint.OpType.POST,
+            path = "set-environment-warning-configuration",
+            description = "Sets the Environment Warning Configuration")
+})
+public class SetEnvironmentWarningConfigurationCommand implements AdminCommand {
+
+    @Inject
+    private Target targetUtil;
+    
+    @Inject
+    EnvironmentWarningService environmentWarningService;
+    
+    @Param(name = "enabled", alias = "Enabled")
+    private Boolean enabled;
+    
+    @Param(name = "message", alias = "Message")
+    private String message;
+    
+    @Param(name = "backgroundColour", alias = "BackgroundColour")
+    private String backgroundColour;
+    
+    @Param(name = "textColour", alias = "TextColour")
+    private String textColour;
+    
+    private final String target = "server-config";
+    
+    @Override
+    public void execute(AdminCommandContext acc) {
+        Config config = targetUtil.getConfig(target);
+        ActionReport actionReport = acc.getActionReport();
+        
+        EnvironmentWarningConfiguration environmentWarningConfiguration = 
+                config.getExtensionByType(EnvironmentWarningConfiguration.class);
+        
+        if (environmentWarningConfiguration != null) {
+            try {
+                ConfigSupport.apply(new SingleConfigCode<EnvironmentWarningConfiguration>(){
+                    @Override
+                    public Object run(EnvironmentWarningConfiguration config) throws PropertyVetoException{
+                        config.setEnabled(enabled);
+                        config.setMessage(message);
```

Comment:
Checks to see that the given colours are correct hex values that can be rendered by the browser.